src/profile-logic/profile-data.js

```diff
@@ -1027,6 +1034,59 @@ export function filterThreadToSearchString(
   );
 }
 
+export function filterThreadByTab(
+  thread: Thread,
+  relevantPages: Set<InnerWindowID>
+): Thread {
+  return timeCode('filterThreadByTab', () => {
+    if (relevantPages.size === 0) {
+      // Either there is no relevant page or "active tab only" view is not active.
+      return thread;
```

Comment:
Optional: It might be worth testing that this properly memoizes, and returns the original thread if no work is needed to be done here. Especially as `getSamplesSelectedStates` takes the result of this as the argument.